addons/kotsadm/alpha/install.sh

```diff
@@ -123,33 +121,23 @@ function kotsadm_secret_cluster_token() {
 function kotsadm_secret_authstring() {
     local AUTHSTRING=$(kubernetes_secret_value default kotsadm-authstring kotsadm-authstring)
 
-    if [ -n "$AUTHSTRING" ]; then
-        # These are the only two valid formats.  Regenerating token in other cases to fix existing installs.
-        if [[ "$AUTHSTRING" =~ ^'Kots ' ]]; then
-            return 0
-        fi
-        if [[ "$AUTHSTRING" =~ ^'Bearer ' ]]; then
-            return 0
-        fi
+    if [[ -z "$AUTHSTRING" || (! "$AUTHSTRING" =~ ^'Kots ' && ! "$AUTHSTRING" =~ ^'Bearer ') ]]; then
```

Comment:
I'm not sure about these order of operations. Is there another way to write this?

Reply:
sure, did test this will multiple scenarios FWIW